File: Angular/TaskManager/src/app/project.ts
import { ClientLocation } from "./client-location";

export class Project {
    projectID:number;
    projectName:string;
    dateOfStart:string;
    teamSize:number;
    active:boolean;
    status:string;
    clientLocationId:number;
    clientLocation:ClientLocation;

    /**
     *
     */
    constructor() {
        this.projectID = 0;
        this.projectName = "";
        this.dateOfStart = Date.now().toString();
        this.teamSize = 0;
        this.active = false;
        this.status = '';
        this.clientLocationId = 0;
        this.clientLocation = new ClientLocation();
    }
}

File: TaskManagerAPI/TaskManagerMvc/Models/Project.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;

namespace TaskManagerMvc.Models
{
	public class Project
	{
		[Key]
		public int ProjectID { get; set; }
		public string ProjectName { get; set; }
		public DateTime DateOfStart { get; set; }
		public int? TeamSize { get; set; }
		public bool Active { get; set; }
		public string Status { get; set; }

		public int ClientLocationId { get; set; }

        [ForeignKey("ClientLocationId")]
		[ValidateNever]
        public virtual ClientLocation ClientLocation { get; set; }


        public Project()
		{
		}
	}

	
}

